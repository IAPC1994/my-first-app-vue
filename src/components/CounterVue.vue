<template>
  <section>
    <h3>Counter: {{ count }}</h3>
    <h3>Square: {{ squareCounter }}</h3>
    <div>
      <button class="btn" @click="count++">+1</button>
      <button class="btn" @click="count--">-1</button>
    </div>
  </section>
</template>

<script lang="ts" setup>
import { useCounter } from '../composables/useCounter';

interface Props {
  value: number;
}

const props = defineProps<Props>();
const { count, squareCounter } = useCounter(props.value);
</script>

<style scoped>
.btn {
  @apply p-5 bg-blue-500 mr-2 rounded hover:bg-blue-700;
}
</style>
