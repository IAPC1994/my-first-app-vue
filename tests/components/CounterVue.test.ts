import { mount } from '@vue/test-utils';
import CounterVue from '@/components/CounterVue.vue';

describe('Testing <CounterVue />', () => {
  test('Should match snapshot', () => {
    const wrapper = mount(CounterVue, {
      props: {
        value: 5,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Renders the counter value correctly', () => {
    const value = 5;
    const wrapper = mount(CounterVue, {
      props: {
        value: value,
      },
    });
    const [counterLabel, squareLabel] = wrapper.findAll('h3');

    expect(wrapper.find('h3').text()).toContain(`Counter: ${value}`);
    expect(wrapper.find('[data-testid = "square-label"]').text()).toContain(
      `Square: ${value * value}`,
    );

    expect(counterLabel.text()).toContain(`Counter: ${value}`);
    expect(squareLabel.text()).toContain(`Square: ${value * value}`);
  });
});
