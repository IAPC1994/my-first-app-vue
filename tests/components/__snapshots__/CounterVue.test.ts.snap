// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing <CounterVue /> > Should match snapshot 1`] = `
"<section data-v-07710f80="">
  <h3 data-v-07710f80="">Counter: 5</h3>
  <h3 data-v-07710f80="" data-testid="square-label">Square: 25</h3>
  <div data-v-07710f80=""><button data-v-07710f80="" class="btn">+1</button><button data-v-07710f80="" class="btn">-1</button></div>
</section>"
`;
